src/rand.rs

```diff
@@ -123,23 +120,82 @@ use self::sysrand_or_urandom::fill as fill_impl;
 #[cfg(any(target_os = "macos", target_os = "ios"))]
 use self::darwin::fill as fill_impl;
 
+#[cfg(target_os = "linux")]
+mod sysrand_chunk {
+    use error;
+    use c;
+
+    extern {
+        fn syscall(number: c::long, ...) -> c::long;
+    }
+
+    #[cfg(target_arch = "aarch64")]
+    const GETRANDOM: c::long = 278;
+    #[cfg(target_arch = "arm")]
+    const GETRANDOM: c::long = 384;
+    #[cfg(target_arch = "x86")]
+    const GETRANDOM: c::long = 355;
+    #[cfg(target_arch = "x86_64")]
+    const GETRANDOM: c::long = 318;
+
+    const EINTR: c::long = 4;
+
+    #[inline]
+    pub fn chunk(dest: &mut [u8]) -> Result<usize, error::Unspecified> {
+        let chunk_len: c::size_t = dest.len();
+        let flags: c::int = 0;
+        let r = unsafe {
+            syscall(GETRANDOM, dest.as_mut_ptr(), chunk_len, flags)
+        };
+        if r < 0 {
+            if r == EINTR {
```

Comment:
This can never be true because `r < 0` and `EINTR == 4`. It should be comparing `errno` instead.